server/src/middleware/errorHandler.ts

```
@@ -0,0 +1,151 @@
+  }
+};
+
+// 404 handler
+export const notFoundHandler = (req: Request, res: Response, next: NextFunction): void => {
+  const message = `Can't find ${req.originalUrl} on this server`;
+  const error = new AppError(message, 404);
+  next(error);
+};
+
+// Async error wrapper
+export const asyncHandler = (fn: Function) => {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)